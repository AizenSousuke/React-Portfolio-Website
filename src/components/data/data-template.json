{
    "name": "yourname",
    "projects": [
        {
            "name": "projectname",
            "description": "projectdescription",
            "link": "https://github.com/"
        },
        {
            "name": "projectname",
            "description": "projectdescription",
            "link": "https://github.com/"
        },
        {
            "name": "projectname",
            "description": "projectdescription",
            "link": "https://github.com/"
        }
    ],
    "social": [
        {
            "link": "https://facebook.com/"
        },
        {
            "link": "https://facebook.com/"
        },
        {
            "link": "https://facebook.com/"
        }
    ]
}