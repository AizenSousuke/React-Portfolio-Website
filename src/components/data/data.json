{
	"name": "Nik",
	"contact": "+65 8666 0407",
	"email": "puteranikdimas@gmail.com",
	"title": "I create web apps to solve business needs",
	"subtitle": "My name is Nik and I'm a web application developer. I am passionate in learning new technologies, building websites, applications, games and also in learning how to fly a fighter jet through flight simulation software.",
	"subtitle_button": "Learn more about me",
	"projects": [
		{
			"id": 1,
			"name": "React Portfolio",
			"description": "A portfolio website created using React JS with Levinson from India in a matter of 3 days via Live Share on Visual Studio Code. We started on the 12th and completed on the 15th of May.",
			"link": "https://github.com/AizenSousuke/React-Portfolio-Website",
			"imageLink": "portfolio"
		},
		{
			"id": 2,
			"name": "Blog",
			"description": "Created a blog with Dot Net Core 3.1 as its backend and React JS as the frontend. I'm the backend developer with 3 other people working on the project. We started March and the project is still pending completion.",
			"link": "https://github.com/AizenSousuke/DotNetCoreReactREST",
			"imageLink": "blog"
		},
		{
			"id": 3,
			"name": "Recipe List",
			"description": "A recipe list created using the Ionic Framework. It is using Firebase as its backend and Imgur API as its image hoster.",
			"link": "https://github.com/AizenSousuke/Angular-Ionic-CRUD",
			"imageLink": "recipe"
		}
	],
	"experiences": [
		{
			"month": "May 2012",
			"header": "Graduated from Republic Polytechnic",
			"subtitle": "Singapore",
			"jobscope": "Designing games with a passion",
			"content": "Diploma with Merit in Game Design | GPA 3.71/4.0"
		},
		{
			"month": "July 2014",
			"header": "Technical Assistant",
			"subtitle": "NCS Pte Ltd",
			"jobscope": "Technical Support in a school environment",
			"content": "Balestier Hill Secondary"
		},
		{
			"month": "October 2015",
			"header": "Armed Security Officer",
			"subtitle": "Aetos Pte Ltd",
			"jobscope": "Checking of vehicles and escorting of personnel",
			"content": "Woodland Checkpoint"
		},
		{
			"month": "September 2018",
			"header": "IT Technical Support Officer",
			"subtitle": "ST Electronics",
			"jobscope": "Airfield Surface Movement and Ground Control System Officer",
			"content": "Changi International Airport"
		},
		{
			"month": "Sept 2019",
			"header": "Performance Tester",
			"subtitle": "NCS Pte Ltd",
			"jobscope": "Testing of Applications and Government Websites",
			"content": "HP Loadrunner"
		},
		{
			"month": "Aug 2020",
			"header": "Solutions Developer",
			"subtitle": "Commerce Online Pte Ltd",
			"jobscope": "Full-Stack development with Microsoft ASP.Net Web Forms",
			"content": "Microsoft C#, Microsoft SQL Server, Microsoft Sharepoint, Microsoft Server"
		}
	],
	"social": [
		{
			"link": "https://instagram.com/"
		},
		{
			"link": "https://facebook.com/"
		},
		{
			"link": "https://github.com/AizenSousuke"
		}
	],
	"about": {
		"profile_picture": "avatar",
		"job_title": "Web App Developer",
		"summary": "I think about code and systems all the time. When I'm not at my day job, I experiment with new web technologies such as Docker and serverless architecture. Currently I'm learning about frontend frameworks like ReactJS, Storybook and NextJS.",
		"signature": "Nik",
		"education": {
			"title": "Republic Polytechnic",
			"description": "Diploma in Game Design",
			"imageLink": "http://loremflickr.com/1024/576/school",
			"location": "Singapore",
			"professionalcert": "Industry Connect Job Ready Software Development Course"
		},
		"other_cert_link": "https://industryconnect.io/Authority/ViewCertificate?guid=620a0eb4-039d-417c-8d82-6d2d68d560eb",
		"expertise": [
			{
				"id": 1,
				"title": "React",
				"description": "React Framework, Ionic, React-Native",
				"imageLink": "https://imgur.com/o1zY60n.png"
			},
			{
				"id": 2,
				"title": "ASP .Net",
				"description": "C#, .Net Core, Microsoft ASP Webforms, Telerik UI Controls",
				"imageLink": "https://imgur.com/0Ceufi8.png"
			},
			{
				"id": 3,
				"title": "Games Development",
				"description": "Unity, Game Maker",
				"imageLink": "https://imgur.com/nSAXbxn.png"
			},
			{
				"id": 4,
				"title": "Video Editing",
				"description": "Adobe After Effects and Premiere Pro",
				"imageLink": "https://imgur.com/KxEKMTP.png"
			},
			{
				"id": 5,
				"title": "Dev Ops",
				"description": "Docker, Github, Azure, Amazon",
				"imageLink": "https://imgur.com/NrCMtF8.png"
			},
			{
				"id": 6,
				"title": "MySQL",
				"description": "Linq, EF Framework",
				"imageLink": "https://imgur.com/My0BGwI.png"
			},
			{
				"id": 7,
				"title": "Flight Simulation",
				"description": "FA-18C",
				"imageLink": "https://imgur.com/esfbHUP.png"
			}
		],
		"skills": [
			{
				"id": 1,
				"icon": "fas fa-cogs",
				"title": "C#",
				"proficiency": 75
			},
			{
				"id": 2,
				"icon": "fas fa-scroll",
				"title": "Typescript",
				"proficiency": 70
			},
			{
				"id": 3,
				"icon": "fas fa-scroll",
				"title": "Javascript",
				"proficiency": 65
			},
			{
				"id": 4,
				"icon": "fab fa-docker",
				"title": "Docker",
				"proficiency": 70
			},
			{
				"id": 5,
				"icon": "fab fa-windows",
				"title": "Windows",
				"proficiency": 95
			}
		]
	}
}
