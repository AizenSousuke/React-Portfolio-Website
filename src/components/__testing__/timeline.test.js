import ReactDOM from "react-dom";
import React from "react";
import data from "../data/data.json";
import { render } from "@testing-library/react";
import Timeline from "../timeline/timeline";
import renderer from "react-test-renderer";

it("it renders correctly", () => {
	const timeline = document.createElement("div");
	ReactDOM.render(<div />, timeline);
	ReactDOM.unmountComponentAtNode(timeline);
});

it("it maps data correctly", () => {
	render(<Timeline data={[data]} />);
});

it("it maps data according to snapshot", () => {
	const timeline = renderer.create(<Timeline data={[data]} />);
	expect(timeline).toMatchSnapshot();
});
