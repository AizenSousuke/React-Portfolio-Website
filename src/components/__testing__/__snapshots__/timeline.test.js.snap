// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it maps data according to snapshot 1`] = `
Array [
  <div
    className="timeline"
    data-testid="timeline"
  >
    <header
      className="timeline-header"
    >
      <span
        className="tag is-medium is-danger"
      >
        Current
      </span>
    </header>
    <div
      className="timeline-item is-danger"
    >
      <div
        className="timeline-marker is-icon is-danger"
      >
        <i
          className="fa fa-flag"
        />
      </div>
      <div
        className="timeline-content fullwidth"
        style={
          Object {
            "minWidth": 500,
          }
        }
      >
        <p
          className="heading"
        >
          May 2012
        </p>
        <div
          className="message is-info"
        >
          <div
            className="message-header"
          >
            <div
              className="column title is-5"
            >
              header
            </div>
            <div
              className="column tag is-link has-text-weight-bold"
            >
              My name is Nik and I'm a web application developer. I am passionate in learning new technologies, building websites, applications, games and also in learning how to fly a fighter jet through flight simulation software.
            </div>
          </div>
          <div
            className="message-body"
          >
            <div
              className="subtitle"
            >
              jobscope
            </div>
            <div
              className="is-size-7"
            >
              Ex anim do id aute aute. Quis anim aliqua eu eiusmod occaecat. Consectetur reprehenderit do commodo do. Occaecat eu magna in deserunt. Qui in aliquip ipsum consequat ipsum ea reprehenderit.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="timeline-header"
    >
      <span
        className="tag is-medium is-danger"
      >
        Past
      </span>
    </div>
  </div>,
  <div
    className="column"
  />,
]
`;
